feat: Block, Tx, TxIn and TxOut struct

rllola opened the pull request:
Added struct Block, Tx, TxIn and TxOut and their associated deserialize functions.

rllola committed:
feat: added Block, Tx, TxIn and TxOut struct with deserialize function;

rllola committed:
feat: added hash and serialize_header function

rllola committed:
fix Tx; added deserialize for TxOut and TxIn;

rllola committed:
remove dbg!

ethicnology reviewed (commented):
hex est-il vraiment nécessaire en dépendency non-dev

rllola commented:
> hex est-il vraiment nécessaire en dépendency non-dev

Non c'est pas nécessaire je peux l'enlever.

rllola commented:
When deserializing we should return a `Result` because it can fail. In that case we should be able to catch it and analyse the bytes that fails.

ethicnology committed:
refactor:

- fix transaction deserialize
- wrap serializers in Result to propagate errors
- simplify code when it's possible

rllola committed:
version deserialize return Result type

rllola committed:
wip

rllola committed:
deserialize return Result; CI rust tests;

rllola committed:
fix message deserialize

rllola committed:
fix coverage

rllola committed:
improve errors; fix typo in CI;

rllola committed:
fix super slow

rllola committed:
remove dbg; remove hex dependency;

rllola committed:
remove some uft8 strings
